packages/medusa/src/api-v2/store/products/validators.ts

```diff
@@ -0,0 +1,49 @@
+import { z } from "zod"
+import {
+  GetProductsParams,
+  ProductStatusEnum,
+} from "../../utils/common-validators"
+import {
+  createFindParams,
+  createOperatorMap,
+  createSelectParams,
+} from "../../utils/validators"
+
+export const StoreGetProductParams = createSelectParams()
+
+export type StoreGetProductVariantsParamsType = z.infer<
+  typeof StoreGetProductVariantsParams
+>
+export const StoreGetProductVariantsParams = createFindParams({
+  offset: 0,
+  limit: 50,
+}).merge(
+  z.object({
+    q: z.string().optional(),
+    id: z.union([z.string(), z.array(z.string())]).optional(),
+    status: ProductStatusEnum.array().optional(),
+    created_at: createOperatorMap().optional(),
+    updated_at: createOperatorMap().optional(),
+    deleted_at: createOperatorMap().optional(),
+    $and: z.lazy(() => StoreGetProductsParams.array()).optional(),
+    $or: z.lazy(() => StoreGetProductsParams.array()).optional(),
+  })
+)
+
+export type StoreGetProductsParamsType = z.infer<typeof StoreGetProductsParams>
+export const StoreGetProductsParams = createFindParams({
+  offset: 0,
+  limit: 50,
+}).merge(
+  z
+    .object({
+      region_id: z.string().optional(),
+      currency_code: z.string().optional(),
+      customer_id: z.string().optional(),
```

Comment:
What exactly does it mean to filter a product by customer id?

Reply:
these are not filters, they're parameters for pricing context to calculate prices

Reply:
Same. Thats a good point about customer_id, I'll use a middleware to pick it up from the token. 

I've added a todo comment there for publishable keys, I don't have the answer for it right now. Do you know if the concept of a default sales channel is being carried over to v2?

Reply:
> Do you know if the concept of a default sales channel is being carried over to v2?

It will be carried over. The store holds a default sales channel ID, similar to v1